transparentpath/gcsutils/transparentpath.py

```diff
@@ -1693,6 +1698,39 @@ def cast_iterable(self, iter_: Iterable):
             to_ret = [self.cast_iterable(item) for item in iter_]
             return to_ret
 
+    def caching_saver(self, file):
+        """
+        save fetched data in tmp file or dict, if not exceeding caching_max_memory else remove oldest data
+        """
+        if self.enable_caching:
+            used_memory = 0
+            if self.caching == "ram":
+                if sys.getsizeof(file) > self.caching_max_memory:
+                    raise BufferError("TOO FAT FILE")
+                for i in self.cached_data_dict.items():
+                    used_memory += sys.getsizeof(i)
+                while used_memory+sys.getsizeof(file) > self.caching_max_memory:
+                    bye = self.cached_data_dict.popitem(last=False)
+                    used_memory -= sys.getsizeof(bye)
+                self.cached_data_dict[self] = file
+            elif self.caching == "tmpfile":
+                temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=self.suffix)
+                self.get(temp_file.name)
+                tempfilesize = temp_file.file.tell()
+                temp_file.close()
+                if tempfilesize > self.caching_max_memory:
```

Comment:
find a way to estimate the memory usage before saving, and same as above do not raise if too big but warn